packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -67,6 +70,19 @@ contract Exchange is
     _;
   }
 
+  // TODO: Exchange migrations need to be initilised with BB address.
+  modifier checkTradingMode() {
+    if (address(breakerBox) != address(0)) {
+      require(
+        breakerBox.checkBreakers(address(this)) == 0,
```

Comment:
Maybe let's define the different mode numbers as named constants, so we're not using magic numbers throughout the code.

Reply:
Also, isn't this a pretty expensive computation we're adding to every trade? I don't think I saw `lastUpdatedTime` used to short circuit if the check was already performed in a block.

Reply:
> Maybe let's define the different mode numbers as named constants, so we're not using magic numbers throughout the code.

It's definitely better to introduce named constants but this check is already expensive. At this stage we only have one breaker and we know if the trading mode is not 0 then don't trade. Is it really worth adding this in?

> Also, isn't this a pretty expensive computation we're adding to every trade? I don't think I saw `lastUpdatedTime` used to short circuit if the check was already performed in a block.

Agreed. This is one of the outstanding things to do. The check will require more than I initially thought but will be updating the implementation to short circuit when possible.